packages/perseus/src/renderer.tsx

```diff
@@ -223,6 +223,8 @@ class Renderer extends React.Component<Props, State> {
     // The i18n linter.
     _translationLinter: TranslationLinter;
 
+    _widgetContainers: Map<string, WidgetContainer> = new Map();
```

Comment:
Hmm... I am not 100% sure how this works in class components, but we should make sure we are not overwriting our refs on re-render. Just something to look out for!